subprojects/core/src/main/java/org/gradle/execution/CompositeAwareTaskSelector.java

```diff
@@ -51,13 +50,9 @@ public CompositeAwareTaskSelector(GradleInternal gradle, BuildStateRegistry buil
                 return getSelectorForChildBuild(build).getFilter(taskPath.removeFirstSegments(1).toString());
             }
         }
-        // Exclusion didn't match an included build, so it might be a subproject of the root build or a relative path
-        if (gradle.isRootBuild()) {
-            return getUnqualifiedBuildSelector().getFilter(path);
-        } else {
-            // Included build ignores this exclusion since it doesn't apply directly to it
-            return Specs.satisfyAll();
-        }
+        return getUnqualifiedBuildSelector().getFilter(path);
```

Comment:
This is going to have consequences with included builds too and common task names like `jar`.

Take a look at this sample:
https://docs.gradle.org/current/samples/sample_sharing_convention_plugins_with_build_logic.html

If you run `gradle help -x jar` with these changes. The build will fail without running the jar task to generate the plugin:
```
* What went wrong:
Could not apply requested plugin [id: 'myproject.java-conventions'] as it does not provide a plugin with id 'myproject.java-conventions'. This is caused by an incorrect plugin implementation. Please contact the plugin author(s).
> Plugin with id 'myproject.java-conventions' not found.
```

Previously, it was not possible to run tasks from an included build, so being able to exclude tasks didn't really mean anything. We also have it so that you cannot use unqualified task selection with included build tasks, so making the fuzzy match work with excludes makes this inconsistent.

Is there some way we could...
1. Allow the StartParameter in the `GradleBuild` task to be configured programmatically such that excludes work. If someone wants to emulate the old behavior, they could pass exclusions to the task.
2. Or make excludes work in a similar way to included builds. So if the GradleBuild task is called "foo", you have to exclude tasks from that build with `-x :foo:someTask`
3. Or make unqualified excludes apply to builds created via `GradleBuild` in the same way as the root build (without affecting other included builds).

Reply:
So... I re-watched some of Adam's Tech Sessions and armed with some more knowledge I tried to dig further. 

I tried to figure out why the `GradleBuild` bits don't show up as an included build when [CompositeAwareTaskSelector.getFilter()](https://github.com/gradle/gradle/blob/ae3758aa7698196c03e7290116ea4b13388986c0/subprojects/core/src/main/java/org/gradle/execution/CompositeAwareTaskSelector.java#L45) is being executed. 

What I've found was that [CompositeAwareTaskSelector.getFilter()](https://github.com/gradle/gradle/blob/ae3758aa7698196c03e7290116ea4b13388986c0/subprojects/core/src/main/java/org/gradle/execution/CompositeAwareTaskSelector.java#L45) runs much sooner (and on a different thread) than [NestedRootBuildRunner.createNestedBuildTree](https://github.com/gradle/gradle/blob/ae3758aa7698196c03e7290116ea4b13388986c0/subprojects/core/src/main/java/org/gradle/internal/build/NestedRootBuildRunner.java#L40). 

Basically, when we try to build the task filters, the `GradleBuild` part has not been evaluated yet, we aren't aware of it yet. Tackling this feels like a much bigger can of worms than the original issue...

@big-guy @adammurdoch , can you guys pls. advise? Should we go further and if so, then how?